### PR TITLE
Deprecate AbstractHlrc* and AbstractHlrcStreamable* base test classes

### PR DESCRIPTION
**Pull request by martijnvg**
* Deprecate AbstractHlrcXContentTestCase and AbstractHlrcStreamableXContentTestCase base test classes in favour for AbstractResponseTestCase.
  The AbstractResponseTestCase tests hlrc serialization in the same way serialization would happen in production. AbstractHlrcXContentTestCase and
  AbstractHlrcStreamableXContentTestCase convert the hlrc instance into a server instance in order to do the comparison.
* Moved AbstractHlrcXContentTestCase and AbstractHlrcStreamableXContentTestCase and subclasses from xpack core to hlrc project.
  This was a bit messy, because licenses had to be changed and sometimes a rename was needed because a test with the same already existsed in hlrc module.
  (This tests also tested parsing, but those tests are not realistic. (hlrc test instance is generated and then parsed hlrc tested compared against that test instance ))
* Removed hlrc test dependency from x-pack core.

### CODE REVIEW

File: client/rest-high-level/src/test/java/org/elasticsearch/client/AbstractHlrcStreamableXContentTestCase.java [hunk 1]
```diff
@@ -14,6 +27,8 @@
 
 import static org.elasticsearch.test.AbstractXContentTestCase.xContentTester;
 
+// TODO: Remove and change subclasses to use AbstractResponseTestCase instead
+@Deprecated
```
Comment (hub-cap) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/AbstractHlrcStreamableXContentTestCase.java]: Can you add a comment w/ @deprecated please use this.class.java instead for our IDE friends.

File: client/rest-high-level/src/test/java/org/elasticsearch/client/AbstractHlrcXContentTestCase.java [hunk 2]
```diff
@@ -1,16 +1,31 @@
...
+package org.elasticsearch.client;
 
 import org.elasticsearch.common.xcontent.ToXContent;
 import org.elasticsearch.common.xcontent.XContentParser;
 import org.elasticsearch.test.AbstractXContentTestCase;
 
 import java.io.IOException;
 
+// TODO: Remove and change subclasses to use AbstractResponseTestCase instead
+@Deprecated
```
Comment (hub-cap) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/AbstractHlrcXContentTestCase.java]: Can you add a comment w/ @deprecated please use this.class.java instead for our IDE friends.

**Review by hub-cap — APPROVED** [review]
looks good, thanks for the work here. One minor thing to address, but i dont need to see it again.

File: client/rest-high-level/src/test/java/org/elasticsearch/client/AbstractHlrcXContentTestCase.java (hunk 2 above)
Comment (martijnvg) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/AbstractHlrcXContentTestCase.java]: pushed: f9f217e911b9971b9e70066d5e70510928a0caa6

File: client/rest-high-level/src/test/java/org/elasticsearch/client/AbstractHlrcStreamableXContentTestCase.java (hunk 1 above)
Comment (martijnvg) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/AbstractHlrcStreamableXContentTestCase.java]: pushed: f9f217e911b9971b9e70066d5e70510928a0caa6

**Pull request merged**